Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -1016,6 +1016,11 @@ void RenderGrid::placeItemsOnGrid(std::optional<LayoutUnit> availableLogicalWidt
 
     performAutoPlacement();
 
+    if (isMasonry()) {
+        m_grid.m_currentGrid.get().autoMajorAxisAutoGridItems.clear();
+        m_grid.m_currentGrid.get().autoMajorAxisAutoGridItems.appendVector(autoMajorAxisAutoGridItems);
```

Comment:
Why not just a setter that takes a Vector&&? But also you should try really hard to avoid having to store this vector of renderers. That always ends in tears.

Reply:
Completely removed this code in next iteration.